### PR TITLE
General Cleanup of DSL

### PR DESCRIPTION
**Pull request by srkukarni**
1. Renamed GeneratorSource to simply Source
2. Added Sink definition for complex sinks

### CODE REVIEW

File: heron/api/src/java/com/twitter/heron/dsl/impl/streamlets/SinkStreamlet.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Set;
+
+import com.twitter.heron.api.topology.TopologyBuilder;
+import com.twitter.heron.dsl.Sink;
+import com.twitter.heron.dsl.impl.BaseStreamlet;
+import com.twitter.heron.dsl.impl.sinks.ComplexSink;
+
+/**
+ * ConsumerStreamlet represents en empty Streamlet that is made up of elements from the parent
+ * streamlet after consuming every element. Since elements of the parents are just consumed
```
Comment (nlu90) [inline heron/api/src/java/com/twitter/heron/dsl/impl/streamlets/SinkStreamlet.java]: Comment needs to be updated

File: heron/api/src/java/com/twitter/heron/dsl/Streamlet.java [hunk 2]
```diff
@@ -163,4 +163,12 @@
...
    */
   void consume(SerializableConsumer<R> consumer);
+
+  /**
+   * Applies the sink's put function to every element of the stream
+   * This function does not return anything.
+   * @param sink The Sink whose put method consumes each element
+   * of this streamlet.
+   */
+  void toSink(Sink<R> sink);
```
Comment (nlu90) [inline heron/api/src/java/com/twitter/heron/dsl/Streamlet.java]: Is the ‘toSink’ method general enough that needed by every streamlet?

Is this new method here for stateful processing?

File: heron/api/src/java/com/twitter/heron/dsl/impl/streamlets/SinkStreamlet.java (hunk 1 above)
Comment (srkukarni) [reply heron/api/src/java/com/twitter/heron/dsl/impl/streamlets/SinkStreamlet.java]: updated

File: heron/api/src/java/com/twitter/heron/dsl/Streamlet.java (hunk 2 above)
Comment (srkukarni) [reply heron/api/src/java/com/twitter/heron/dsl/Streamlet.java]: The toSink is just for abstracting out complex sinks(aka terminator bolts) that might need a setup/cleanup.Not really related to stateful processing.

**Pull request merged**